Detectors/PHOS/calib/PHOSCalibWorkflow/src/PHOSEnergyCalibDevice.cxx

```diff
@@ -0,0 +1,107 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "PHOSCalibWorkflow/PHOSEnergyCalibDevice.h"
+#include "DataFormatsPHOS/TriggerRecord.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "CCDB/CcdbApi.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DetectorsCalibration/Utils.h"
+#include "Framework/ControlService.h"
+
+#include "FairLogger.h"
+#include <string>
+
+using namespace o2::phos;
+
+void PHOSEnergyCalibDevice::init(o2::framework::InitContext& ic)
+{
+  mCalibrator.reset(new PHOSEnergyCalibrator());
+
+  //TODO!!! read configuration
+  // mPtMin,mEminHGTime,mEminLGTime
+
+  //Configure output Digits file
+  mCalibrator->setOutDigitsFile(mdigitsfilename);
+
+  //read calibration and bad map objects and send them to calibrator
+  if (!mBadMap) {
+    if (mUseCCDB) {
+      LOG(INFO) << "Retrieving BadMap from CCDB";
+      o2::ccdb::CcdbApi ccdb;
+      ccdb.init(mCCDBPath); // or http://localhost:8080 for a local installation
+      std::map<std::string, std::string> metadata;
+      mBadMap.reset(ccdb.retrieveFromTFileAny<BadChannelMap>("PHS/Calib/BadChannels", metadata, mRunStartTime));
```

Comment:
just for the record: at the moment it is ok, but in future the manager will own the objects. Also, if you would use `BasicCCDBManager`, which supports caching, you should not take the ownership of objects.

Reply:
Thanks, lets update this once framework will be ready